src/aliceVision/image/io.cpp

```diff
@@ -165,26 +193,39 @@ void writeImage(const std::string& path,
                 oiio::TypeDesc typeDesc,
                 int nchannels,
                 const Image<T>& image,
+                EImageColorSpace imageColorSpace,
                 const oiio::ParamValueList& metadata = oiio::ParamValueList())
 {
   const fs::path bPath = fs::path(path);
   const std::string extension = bPath.extension().string();
   const std::string tmpPath =  (bPath.parent_path() / bPath.stem()).string() + "." + fs::unique_path().string() + extension;
   const bool isEXR = (extension == ".exr");
+  //const bool isTIF = (extension == ".tif");
+  const bool isJPG = (extension == ".jpg");
+  const bool isPNG = (extension == ".png");
 
   oiio::ImageSpec imageSpec(image.Width(), image.Height(), nchannels, typeDesc);
   imageSpec.extra_attribs = metadata; // add custom metadata
 
-  if(isEXR)
+  if((isJPG || isPNG) && imageColorSpace != EImageColorSpace::NO_CONVERSION)
```

Comment:
Add `EImageColorSpace::AUTO` to decide based on the exception.
Throw an exception if the colorspace is incoherent with the file extension.